src/sagas.js

```diff
@@ -59,59 +59,59 @@ export function* updateUrl(): Saga {
   window.location.hash = localUrlFromVersion(state.version);
 }
 
-// Refreshing a status for the current version.
-export function* refreshStatus(): Saga {
+export function* checkResultAndUpdateAndNotify(
+  title: string,
+  url: string,
+  prevResult: CheckResult,
+): Saga {
   const notifyChanges = (checkTitle, status) => {
     if (Notification.permission === 'granted') {
       new Notification(`${checkTitle}: status changed (${status}).`);
     }
   };
 
+  yield call(checkResultAndUpdate, title, url);
+  const state: State = yield select();
+  const result: CheckResult = state.checkResults && state.checkResults[title];
+  if (prevResult && result && prevResult.status !== result.status) {
+    notifyChanges(title, result.status);
+  }
+}
+
+// Refreshing a status for the current version.
+export function* refreshStatus(): Saga {
   const state: State = yield select();
   // Save previous results so we can check if something changed.
   const prevResults = state.checkResults;
   yield put(setVersion(state.version));
-  if (state.releaseInfo) {
-    let checks = {};
-    state.releaseInfo.checks.map(
-      ({url, title}) => (checks[title] = call(checkStatus, url)),
+  if (state.releaseInfo && state.releaseInfo.checks) {
+    yield state.releaseInfo.checks.map(({url, title}) =>
```

Comment:
Do we not need something like `yield.all`? How does this work?

Reply:
I don't remember if it's documented in the official docs, but: https://github.com/redux-saga/redux-saga/issues/306#issuecomment-217141546

Reply:
Doesn't this say `yield all(foo.map(...))`?

Reply:
wait, it does. How does this thing even work then... this is beyond me. I'll patch on master, and hope for the best. This is smelly...

Reply:
See #94